pkg/networkdevice/metadata/payload_utils_test.go

```diff
@@ -0,0 +1,73 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package metadata
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/snmp/common"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func Test_batchPayloads(t *testing.T) {
+	collectTime := common.MockTimeNow()
+	deviceID := "123"
+	devices := []DeviceMetadata{
+		{ID: deviceID},
+	}
+
+	var interfaces []InterfaceMetadata
+	for i := 0; i < 350; i++ {
+		interfaces = append(interfaces, InterfaceMetadata{DeviceID: deviceID, Index: int32(i)})
+	}
+	var ipAddresses []IPAddressMetadata
+	for i := 0; i < 100; i++ {
+		ipAddresses = append(ipAddresses, IPAddressMetadata{InterfaceID: deviceID + ":1", IPAddress: "1.2.3.4", Prefixlen: 24})
+	}
+	var topologyLinks []TopologyLinkMetadata
+	for i := 0; i < 100; i++ {
+		topologyLinks = append(topologyLinks, TopologyLinkMetadata{
+			Local:  &TopologyLinkSide{Interface: &TopologyLinkInterface{ID: "a"}},
+			Remote: &TopologyLinkSide{Interface: &TopologyLinkInterface{ID: "b"}},
+		})
+	}
+	payloads := BatchPayloads("my-ns", "127.0.0.0/30", collectTime, 100, devices, interfaces, ipAddresses, topologyLinks)
+
+	assert.Equal(t, 6, len(payloads))
+
+	assert.Equal(t, "my-ns", payloads[0].Namespace)
+	assert.Equal(t, "127.0.0.0/30", payloads[0].Subnet)
+	assert.Equal(t, int64(946684800), payloads[0].CollectTimestamp)
+	assert.Equal(t, devices, payloads[0].Devices)
+	assert.Equal(t, 99, len(payloads[0].Interfaces))
```

Comment:
Assert has a `.Len` helpers for this with a better error message is the test fails. Same goes for all the `len()` tests bellow.

Reply:
👍 , fixed here https://github.com/DataDog/datadog-agent/pull/16359/commits/0cae21a464fbfd5e536523f6feb7bedcd55db941